ark/jsonschema/composition.ts

```diff
@@ -0,0 +1,60 @@
+import { printable } from "@ark/util"
+import { type, type JsonSchema, type Type } from "arktype"
+import { parseJsonSchema } from "./json.ts"
+
+const parseAllOfJsonSchema = (jsonSchemas: readonly JsonSchema[]): Type =>
+	jsonSchemas
+		.map(jsonSchema => parseJsonSchema(jsonSchema))
+		.reduce((acc, validator) => acc.and(validator))
+
+const parseAnyOfJsonSchema = (jsonSchemas: readonly JsonSchema[]): Type =>
+	jsonSchemas
+		.map(jsonSchema => parseJsonSchema(jsonSchema))
+		.reduce((acc, validator) => acc.or(validator))
+
+const parseNotJsonSchema = (jsonSchema: JsonSchema) => {
```

Comment:
Should use `: Type` and `as never` here or similar so we have explicit return type annotations. Doesn't matter here really because we don't export it, but just a good pattern anyways as it makes the expectation clearer

Reply:
Fixed in 984f0104